8.0 portage of module stock_split_picking

yvaucher opened the pull request.

yvaucher committed:
Reactivate stock_split_picking

yvaucher committed:
remove stock.picking.out

yvaucher committed:
Port to 8.0

yvaucher committed:
Update translations

yvaucher committed:
README template, summary, contributors and icon

pedrobaeza commented on stock_split_picking/__openerp__.py:
```diff
@@ -18,28 +18,19 @@
  'author': "Camptocamp,Odoo Community Association (OCA)",
  'maintainer': 'Camptocamp',
- 'category': 'stock',
+ 'category': 'Warehouse',
```
Warehouse management

pedrobaeza commented:
You can consider adding standard OCA's icon for module: https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.png

gurneyalex commented on stock_split_picking/README.rst:
```diff
@@ -0,0 +1,46 @@
+
+This module adds a "Split" button on the outgoing pickings form.
+
+It works like the classical picking Transfer but it does it let
```
s/it does it let/it leaves/

yvaucher commented:
@pedrobaeza Nice new icon

yvaucher committed:
Adapt tests

guewen commented on stock_split_picking/model/stock.py:
```diff
@@ -0,0 +1,49 @@
+class stock_picking(models.Model):
+    """Adds picking split without done state.
+    Long piece of code has to be copied from OpenERP
+    as there is no hook..."""
```
This docstring can be removed

yvaucher replied:
Right thanks

yvaucher commented:
@guewen fixed

guewen commented on stock_split_picking/README.rst:
```diff
@@ -0,0 +1,45 @@
+======================
+
+ * When splitting a picking in an unassigned state, wizard won't
+   be auto completed with picking lines.
```
It could be an uncomfortable limitation, because I guess that most of the time we split a picking because lines are not available.

Commit:
Split procurements as well when a move is split

Both procurements will still be split to the sale order line.
If one of the picking is canceled, the whole sale order will be in
'shipping_exception'.

Commit:
Allow to split a partially available picking

jgrandguillaume commented:
Hi,

Red build a-side, I've tested it and it works fine ! The only issue is that when the goods are not available, you need to enter manually the quantity to be split because the "split wizard" is empty. If goods are available it's fine. I think we should go this way and see if an improvement will be needed to better handle this case.

So once the build are green :+1: for me

Regards,

Joël

gurneyalex committed:
also show split button on incoming moves

lepistone commented on stock_split_picking/test/test_picking_split.yml:
```diff
@@ -0,0 +1,69 @@
+  I check that the backorder has 40 units with state set to assigned not done
+-
+  !python {model: stock.picking, id: outgoing_shipment_gameboy}: |
+    self = self.search([('backorder_id', '=', self.id)])
```
Why do you reassign `self`? Could you use another variable instead?

lepistone commented:
:+1: except a minor remark in a test. @guewen OK for you to merge even with current limitations?

gurneyalex commented:
@yvaucher added a fixup commit to explain what was annoying @lepistone 

If this suits you both, then :+1:

yvaucher commented:
@gurneyalex ok for me

guewen commented:
@lepistone yep, ok for me

gurneyalex commented:
I found an issue with this PR (and I'm trying to fix it): the reservations are not split

gurneyalex committed:
handle reservations

when splitting a move with reservations, they have to be recomputed.
code inspired from the implementation of stock_picking.rereserve_quants

split the tests in travis.yml because another module is changing the states we get
with only stock_split_picking installed

fix a bug in stock_ownership_by_move which would cause wrong pack operations to
be generated

gurneyalex commented:
Issue with reservations fixed. :+1:

lepistone commented:
:tada:

yvaucher commented:
:+1:

lepistone commented:
Test failure is probably unrelated. I restarted the build. I'll merge if it goes well.

yvaucher commented:
@lepistone a rebase is needed can you do it ?

yvaucher commented:
Rebased to fix conflict in travis config

lepistone committed:
fix tests: mark pickings as Todo

lepistone commented:
I fixed a bug in the tests of `stock_ownership_by_move`. Let's merge once green.

yvaucher commented:
One job fail with `ImportError: No module named pytz`

I restarted it

yvaucher commented:
Ok it's green
